File: src/main/java/me/zeroeightsix/kami/module/modules/bewwawho/misc/FormatChat.java
package me.zeroeightsix.kami.module.modules.bewwawho.misc;

import me.zeroeightsix.kami.module.Module;
import me.zeroeightsix.kami.event.events.PacketEvent;
import me.zeroeightsix.kami.command.Command;
import me.zeroeightsix.kami.util.zeroeightysix.Wrapper;

import me.zero.alpine.listener.EventHandler;
import me.zero.alpine.listener.Listener;

import net.minecraft.client.Minecraft;
import net.minecraft.network.play.client.CPacketChatMessage;

/**
 * Created on 16 December by 0x2E | PretendingToCode
 */

@Module.Info(name = "FormatChat", description = "Add color and linebreak support to upstream chat packets", category = Module.Category.MISC)
public class FormatChat extends Module {

    @Override
    public void onEnable() {
        if (Minecraft.getMinecraft().getCurrentServerData() == null) {
            Command.sendWarningMessage("[FormatChat] &6&lWarning: &r&6This does not work in singleplayer");
            this.disable();
        }
        else {
            Command.sendWarningMessage("[FormatChat] &6&lWarning: &r&6This will kick you on most servers!");
        }
    }

    @EventHandler
    public Listener<PacketEvent.Send> sendListener = new Listener<>(event -> {
        if (event.getPacket() instanceof CPacketChatMessage) {
            String message = ((CPacketChatMessage) event.getPacket()).message;
            if(message.contains("&") || message.contains("#n")){
                message = message.replaceAll("&", Command.SECTION_SIGN + "");
                message = message.replaceAll("#n", "\n");

                Wrapper.getPlayer().connection.sendPacket(new CPacketChatMessage(message));
                event.cancel();
            }
        }
    });
}

File: src/main/kotlin/com/lambda/client/module/modules/chat/FormatChat.kt
package com.lambda.client.module.modules.chat

import com.lambda.client.manager.managers.MessageManager.newMessageModifier
import com.lambda.client.module.Category
import com.lambda.client.module.Module
import com.lambda.client.util.text.MessageSendHelper

internal object FormatChat : Module(
    name = "FormatChat",
    description = "Add color and linebreak support to upstream chat packets",
    category = Category.CHAT,
    modulePriority = 300
) {
    private val modifier = newMessageModifier {
        it.packet.message
            .replace('&', '§')
            .replace("#n", "\n")
    }

    init {
        onEnable {
            if (mc.currentServerData == null) {
                MessageSendHelper.sendWarningMessage("$chatName &6&lWarning: &r&6This does not work in singleplayer")
                disable()
            } else {
                MessageSendHelper.sendWarningMessage("$chatName &6&lWarning: &r&6This will kick you on most servers!")
                modifier.enable()
            }
        }

        onDisable {
            modifier.enable()
        }
    }
}